sdk/storage/azure-storage-blob-v2/src/main/java/com/azure/v2/storage/blob/ContainerClient.java

```diff
@@ -0,0 +1,1065 @@
+        this.serviceClient.create(containerName, timeout, metadata, access, requestId, blobContainerEncryptionScope);
+    }
+
+    /**
+     * returns all user-defined metadata and system properties for the specified container. The data returned does not
+     * include the container's list of blobs.
+     * 
+     * @param containerName The container name.
+     * @param timeout The timeout parameter is expressed in seconds. For more information, see &lt;a
+     * href="https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/setting-timeouts-for-blob-service-operations"&gt;Setting
+     * Timeouts for Blob Service Operations.&lt;/a&gt;.
+     * @param leaseId If specified, the operation only succeeds if the resource's lease is active and matches this ID.
+     * @param requestId Provides a client-generated, opaque value with a 1 KB character limit that is recorded in the
+     * analytics logs when storage analytics logging is enabled.
+     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws StorageErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the response.
+     */
+    @Metadata(generated = true)
+    public Response<Void> getPropertiesWithResponse(String containerName, Integer timeout, String leaseId,
+        String requestId, RequestOptions requestOptions) {
+        return this.serviceClient.getPropertiesWithResponse(containerName, timeout, leaseId, requestId, requestOptions);
+    }
+
+    /**
+     * returns all user-defined metadata and system properties for the specified container. The data returned does not
+     * include the container's list of blobs.
+     * 
+     * @param containerName The container name.
+     * @param timeout The timeout parameter is expressed in seconds. For more information, see &lt;a
+     * href="https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/setting-timeouts-for-blob-service-operations"&gt;Setting
+     * Timeouts for Blob Service Operations.&lt;/a&gt;.
+     * @param leaseId If specified, the operation only succeeds if the resource's lease is active and matches this ID.
+     * @param requestId Provides a client-generated, opaque value with a 1 KB character limit that is recorded in the
+     * analytics logs when storage analytics logging is enabled.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws StorageErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    @Metadata(generated = true)
+    public void getProperties(String containerName, Integer timeout, String leaseId, String requestId) {
```

Comment:
Is there a particular reason why this would return void? Is would be great to see if we can autogenerate a return value here as this [API](https://learn.microsoft.com/en-us/rest/api/storageservices/get-blob-properties?tabs=microsoft-entra-id) returns quite a bit of information, it is just all in the response headers.

Reply:
Noting our conversation here. It is because the `getProperties` is a HEAD request and there is no body to model. We will need to manually add our own strongly typed class especially for backwards compatibility.